### DISCUSSION TITLE
Project to build RISC-V Docker images

### DISCUSSION THREAD

**Discussion by olof-nord** [mastodon/mastodon#27802] [General]
### Pitch

I propose to extend the built docker images to a new architecture, `linux/riscv64`.

Due to the already existing docker buildx usage, the docker part should already be handled.

Ruby already supports RISC-V, Ubuntu supports the riscv64 architecture since the release of Ubuntu 20.04 LTS.

There are not yet any official Node.js builds for risc-v, but unofficial builds are built since about 7 months and published to the unofficial-builds.nodejs.org archive.
https://unofficial-builds.nodejs.org/download/release/v17.7.1/

With some smaller adjustments to the Dockerfile and to the build-image GitHub pipeline, I think supporting this architecture is doable.

### Motivation

RISC-V is an open standard instruction set architecture, which is under rapid development. Distributions including AOSP, Arch Linux, Debian, Gentoo Linux, Fedora, OpenEuler, etc are also actively doing porting work for this architecture.

Unlike most other ISA designs, RISC-V is provided under open source licenses that do not require fees to use.

**Comment by ineffyble** [comment]
@olof-nord I'm curious, do you have a practical application for this in mind? E.g. some hardware with a RISC-V CPU?

**Comment by olof-nord** [comment]
Very valid question.

I really like the architecture for the same reason I like open source: I would argue that the risc-v processor architecture is the closest we have got so far when it comes to open source hardware.

Additionally, I own a riscv64 system myself, a SiFive Unmatched.

With any new architecture, there is a bit of a chicken-egg situation: what should come first. The hardware relies on the software, the software on the hardware. 

As for hardware which is for sale at the moment, from what I can tell the StarFive VisionFive2 is the best option currently.

**Comment by olof-nord** [comment]
I find this topic still worthwhile discussing and to put energy and resources into.
As for hardware, now the Framework team has released a model with RISC-V support.

https://frame.work/de/de/blog/introducing-a-new-risc-v-mainboard-from-deepcomputing
